### PR TITLE
REF don't make as many statuses

### PR DESCRIPTION
**Pull request by beckermr**
Right now we make what appears to be more than one status per commit. This PR puts in a check to make sure only make a new status if it is different or we have not done it before.

Checklist
* [x] Pushed the branch to main repo
* [x] CI passed on the branch

### CODE REVIEW

File: conda_forge_webservices/linting.py [hunk 1]
```diff
@@ -223,16 +223,34 @@ def set_pr_status(owner, repo_name, lint_info, target_url=None):
...
+            if status.context == "conda-forge-linter":
+                last_state = status.state
+
+        # convert the linter status to a state
+        lint_status_to_state = {"good": "success", "mixed": "success"}
+        lint_last_state = lint_status_to_state.get(lint_info['status'], "failure")
+
+        # make a status only if it is different or we have not ever done it
+        # for this commit
+        if last_state is None or last_state != lint_last_state:
```
Comment (isuruf) [inline conda_forge_webservices/linting.py]: This should check that the target URLs are different as well.

Comment (beckermr) [reply conda_forge_webservices/linting.py]: Why?

Comment (isuruf) [reply conda_forge_webservices/linting.py]: For eg: if the commit had two bad statuses.
1. lint failure due to smithy failure - comment 1
2. actual lint - comment 2

Both comments have the same state, but the target_url is different.

Comment (beckermr) [reply conda_forge_webservices/linting.py]: Ohhhh. A bad status should point back to the most recent comment with the correct lints. Will fix.

**Comment by jakirkham** [comment]
Thanks Matt! 😀

**Pull request merged**